### ISSUE TITLE
patternProperties in editor

### ISSUE THREAD

**Issue by svsolopov** [json-editor/json-editor#811]
Object with patternProperties not show in editor

Direct link to example: https://json-editor.github.io/json-editor/

```javascript
{
  "type": "object",
  "patternProperties": {
    "^[0-9]+$": {
      "type": "object",
      "properties": {
        "code": {
          "type": "string",
          "pattern": "^[01]{1}$"
        }
      },
      "required": [
        "code"
      ]
    }
  },
  "additionalProperties": false
}
```

Valid result
 {"0":{"code":"1"},"1":{"code":"1"},}

**Comment by schmunk42** [comment]
Could you elaborate further? I do not understand the problem.

**Comment by germanbisurgi** [comment]
hi @svsolopov 

1) correct format of json data (there is an extra comma at the end)

```
{
	"0": {
		"code": "1"
	},
	"1": {
		"code": "1"
	}
}
```

2) use "additionalProperties": true

**Comment by theotherwillembotha** [comment]
I suspect what @svsolopov  meant was that if one is to use "additionalProperties": true it allows for creating arbitrary properties, not just a property that fall within the defined patternProperties.

The original intention may have been that the "keys" are only allowed to be strings consisting of numbers, and all other keys (apart from those specified in the properties section) are invalid.

**Comment by germanbisurgi** [comment]
The don#t display because there are no properties defined in the schema and "additionalProperties" is set to false. If you set it to true and add properties with number keys it works

**Comment by theotherwillembotha** [comment]
I think my explanation is unclear.

This is the schema:
```
{
  "type": "object",
  "patternProperties": {
    "^[0-9]+$": {
      "type": "object",
      "properties": {
        "code": {
          "type": "string"
        }
      },
      "required": [
        "code"
      ]
    }
  },
  "additionalProperties": true
}
```
And this is the JSON.
```
{
  "12": {
    "code": "bang"
  },
  "16": {
    "code": "pop"
  },
  "foo": "fizz"
}
```
"foo" is valid because we allow additional properties, but it shouldn't be because it doesn't match the pattern "^[0-9]+$"

**Comment by germanbisurgi** [comment]
Now is clear, and yes the logic should be completed, based on this spec https://json-schema.org/understanding-json-schema/reference/object.html

**Comment by robocoder** [comment]
(Kindly ignore my commits referencing this issue.)

`additionalProperties` needs to be `true` to allow properties that aren't explicit `properties`.

The proper fix would be to support `propertyNames` (introduced in version 6 of the spec).

**Comment by schmunk42** [comment]
> The proper fix would be to support propertyNames (introduced in version 6 of the spec).

PRs welcome ;) we currently only fully support up to v4

**Comment by robocoder** [comment]
Fixed by #915
